File: 2507.cc
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#define EPS 1e-5
int main() {
  double x, y, c;
  while (~scanf("%lf%lf%lf", &x, &y, &c)) {
    double l = 0, mid;
    double r = x < y ? x : y;
    double xx, yy, tt;
    while (1) {
      mid = (l + r) / 2;
      xx = sqrt(x * x - mid * mid);
      yy = sqrt(y * y - mid * mid);
      tt = xx * yy / (xx + yy);
      if (fabs(tt - c) < EPS)
        break;
      if (tt < c)
        r = mid;
      else
        l = mid;
    }
    printf("%.3f\n", mid);
  }
  return 0;
}

File: 2507.java
//* @author: 82638882@163.com
import java.io.*;
import java.util.Arrays;
public class Main
{
 public static void main(String[] args) throws IOException
  {
  InputStreamReader is=new InputStreamReader(System.in);
  BufferedReader in=new BufferedReader(is);
  while(in.ready())
   {
	String[] ss=in.readLine().split(" ");
	double x=Double.parseDouble(ss[0]);
	double y=Double.parseDouble(ss[1]);
	double c=Double.parseDouble(ss[2]);
	double max=Math.min(x, y),min=0;
	for(int i=0;max-min>0.00001;i++)
	{
         double mid=(min+max)/2;
	 double d1=Math.sqrt(x*x-mid*mid);
	 double d2=Math.sqrt(y*y-mid*mid);
	 if(c*(d1+d2)==(d1*d2)){
		min=max=mid;
	        break;
	 }
	else if(c*(d1+d2)>(d1*d2)) max=mid;
	else min=mid;
       }
       System.out.printf("%.3f\n",(min+max)/2);
    }
  }
}

